File: scripts/lib/hook_output.py
"""
Hook Output - Dual-channel output system for hooks

Provides functions to output to terminal (stderr) and Claude context (stdout)
based on verbosity configuration.
"""

import sys

try:
    from .config_reader import get_verbosity, should_use_emoji
except ImportError:
    from config_reader import get_verbosity, should_use_emoji

try:
    from .debug_logger import log_debug
except ImportError:
    from debug_logger import log_debug


def to_terminal(message, min_level=2):
    """Output to terminal (stderr) based on verbosity level

    Args:
        message: Message to display
        min_level: Minimum verbosity level to display (default: 2)
    """
    verbosity = get_verbosity()

    # LOG: Trace every call
    log_debug("ANY", "hook_output", f"to_terminal(min_level={min_level}, verbosity={verbosity})", "TRACE")

    if verbosity >= min_level:
        log_debug("ANY", "hook_output", f"Writing to stderr: {message[:100]}...", "OUTPUT")
        sys.stderr.write(message + "\n")
        sys.stderr.flush()
    else:
        log_debug("ANY", "hook_output", f"Skipped (verbosity {verbosity} < {min_level})", "SKIP")


def to_context(message, prefix="CONTEXT"):
    """Output to Claude context (stdout)

    Args:
        message: Message to inject into Claude's context
        prefix: Prefix (CONTEXT, INSTRUCTION, etc.)
    """
    log_debug("ANY", "hook_output", f"to_context(prefix={prefix}, message_len={len(message)})", "TRACE")
    log_debug("ANY", "hook_output", f"Writing to stdout: {prefix}: {message[:100]}...", "OUTPUT")

    sys.stdout.write(f"{prefix}: {message}\n")
    sys.stdout.flush()


def to_both(message, min_level=2, prefix="CONTEXT"):
    """Output to both terminal and context

    Args:
        message: Message to output
        min_level: Minimum verbosity level for terminal
        prefix: Prefix for context injection
    """
    to_terminal(message, min_level)
    to_context(message, prefix)

File: backups/python-legacy-20260112/scripts/lib/hook-output.js
/**
 * Hook Output - Dual-channel output system for hooks
 *
 * Provides functions to output to terminal (stderr) and Claude context (stdout)
 * based on verbosity configuration.
 */

const { getVerbosity, shouldUseEmoji } = require('./config-reader.js');
const { logDebug } = require('./debug-logger.js');

/**
 * Output to terminal (stderr) based on verbosity level
 *
 * @param {string} message - Message to display
 * @param {number} minLevel - Minimum verbosity level to display (default: 2)
 */
function toTerminal(message, minLevel = 2) {
    const verbosity = getVerbosity();

    // LOG: Trace every call
    logDebug('ANY', 'hook-output', `toTerminal(minLevel=${minLevel}, verbosity=${verbosity})`, 'TRACE');

    if (verbosity >= minLevel) {
        logDebug('ANY', 'hook-output', `Writing to stderr: ${message.substring(0, 100)}...`, 'OUTPUT');
        console.error(message);
    } else {
        logDebug('ANY', 'hook-output', `Skipped (verbosity ${verbosity} < ${minLevel})`, 'SKIP');
    }
}

/**
 * Output to Claude context (stdout)
 *
 * @param {string} message - Message to inject into Claude's context
 * @param {string} prefix - Prefix (CONTEXT, INSTRUCTION, etc.)
 */
function toContext(message, prefix = 'CONTEXT') {
    logDebug('ANY', 'hook-output', `toContext(prefix=${prefix}, message_len=${message.length})`, 'TRACE');
    logDebug('ANY', 'hook-output', `Writing to stdout: ${prefix}: ${message.substring(0, 100)}...`, 'OUTPUT');

    console.log(`${prefix}: ${message}`);
}

/**
 * Output to both terminal and context
 *
 * @param {string} message - Message to output
 * @param {number} minLevel - Minimum verbosity level for terminal
 * @param {string} prefix - Prefix for context injection
 */
function toBoth(message, minLevel = 2, prefix = 'CONTEXT') {
    toTerminal(message, minLevel);
    toContext(message, prefix);
}

module.exports = { toTerminal, toContext, toBoth, getVerbosity, shouldUseEmoji };
